projects/novo-elements/src/elements/data-table/cell-headers/data-table-header-cell.component.ts

```diff
@@ -210,6 +293,51 @@ export class NovoDataTableCellHeader<T> implements IDataTableSortFilter, OnInit,
     });
   }
 
+  public isSelected(option, optionsList) {
+    if (optionsList) {
+      const optionValue = option.hasOwnProperty('value') ? option.value : option;
+
+      let found = optionsList.find((item) => {
+        if (item.hasOwnProperty('value')) {
+          return item.value === optionValue;
+        } else {
+          return item === optionValue;
+        }
+      });
+      return found !== undefined;
+    }
+    return false;
+  }
+
+  public toggleSelection(option) {
+    const optionValue = option.value ? option.value : option;
+
+    let optionIndex = this.multiSelectedOptions.findIndex((item) => {
+      if (item.hasOwnProperty('value')) {
+        return item.value === optionValue;
+      } else {
+        return item === optionValue;
+      }
+    });
```

Comment:
looks like  repeated code from lines 300-308

Reply:
The checks inside the find and findIndex are pretty much identical. Would you like for me to externalize this check in another function?